spec-tests/src/test/java/org/candlepin/spec/pools/RefreshPoolsTest.java

```diff
@@ -0,0 +1,877 @@
+        verifyCertProductsAndContent(body, Map.of("content_access", List.of(updatedContent)));
+    }
+
+    @Test
+    @OnlyInHosted
+    void shouldRegenerateEntitlementsWhenProductsForAnEntitledPoolChanges() {
+        ContentDTO content = adminClient.hosted().createContent(Content.random());
+        ProductDTO prod = adminClient.hosted().createProduct(Products.random());
+        prod = adminClient.hosted().addContentToProduct(prod.getId(), content.getId(), true);
+        adminClient.hosted().createSubscription(Subscriptions.random(owner, prod));
+
+        refreshPools(adminClient, owner.getKey());
+        List<PoolDTO> pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+
+        // Verify the product exits in its initial state
+        ProductDTO actualProd = adminClient.ownerProducts().getProductByOwner(ownerKey, prod.getId());
+        assertEquals(prod.getName(), actualProd.getName());
+
+        // Consume the pool so we have an entitlement
+        ConsumerDTO user = Consumers.random(owner, ConsumerTypes.System);
+        user.setUsername("admin");
+        user = adminClient.consumers().createConsumer(user, "admin",
+        ownerKey, null, true);
+        ApiClient consumerClient = ApiClients.trustedConsumer(user.getUuid());
+
+        JsonNode ents = consumerClient.consumers().bindPool(user.getUuid(), pools.get(0).getId(), 1);
+        assertNotNull(ents);
+        assertEquals(1, ents.size());
+        user = adminClient.consumers().getConsumer(user.getUuid());
+        assertEquals(1, user.getEntitlementCount());
+
+        JsonNode entCert = ents.get(0).get("certificates").get(0);
+        JsonNode entSerial = entCert.get("serial").get("serial");
+
+        // Modify the product for this sub
+        String updatedName = StringUtil.random("update-");
+        prod.setName(updatedName);
+        adminClient.hosted().updateProduct(prod.getId(), prod);
+
+        refreshPools(adminClient, owner.getKey());
+        pools = adminClient.pools().listPoolsByOwner(owner.getId());
+        assertEquals(1, pools.size());
+
+        // Verify the product change has been pulled down
+        actualProd = adminClient.ownerProducts().getProductByOwner(ownerKey, prod.getId());
+        assertNotNull(actualProd);
+        assertEquals(updatedName, actualProd.getName());
+
+        // Verify the entitlement cert has changed as a result
+        EntitlementDTO updatedEnt = adminClient.entitlements().getEntitlement(ents.get(0).get("id").asText());
+        assertNotNull(updatedEnt);
+        assertEquals(1, updatedEnt.getCertificates().size());
+        CertificateDTO updatedCert = updatedEnt.getCertificates().stream()
+            .collect(MoreCollectors.onlyElement());
+        assertNotEquals(entSerial.asText(), updatedCert.getSerial().getSerial());
+    }
+
+    @Test
+    @OnlyInHosted
+    void shouldRegenerateEntitlementsWhenRequiredProductsChanges() throws Exception {
```

Comment:
This is a pretty large test and I wasn't sure if it could be broken up or not and so I kept it the same so I didn't lose any test coverage on functionality. However, if someone knows a good way of breaking this up, that would be great.

Original test:
https://github.com/candlepin/candlepin/blob/master/spec/refresh_pools_spec.rb#L642